src/components/ViewTopics.jsx

```diff
@@ -0,0 +1,55 @@
+import { useState, useEffect } from "react";
+import { getTopics } from "../utils/ApiCalls";
+
+import { useNavigate } from "react-router-dom";
+
+const ViewTopics = ({ onTopicChange }) => {
+  const [topics, setTopics] = useState([]);
+  const [selectedTopic, setSelectedTopic] = useState(null);
+  const [loading, setLoading] = useState(false);
```

Comment:
The default value being set to false here is causing the dropdown to appear for a split second then it disappears and re-appears because of your useEffect, but if you set this value to be true by default this should only show the drop down once the request has been made successfully.